### PR TITLE
Automatically deploy backend api based on branch update.

### PR DESCRIPTION
**Pull request by Tadjaur**
- [X] Automatically deploy backend to preview when we push on `andrew_testing` branch
- [X] Automatically deploy backend to production when we push on `main` branch

I made sure it worked before creating the pr: [Here is the run :point_down: ](https://github.com/andrew-bierman/PackRat/actions/runs/10765967688)

Additional Information:
- All sensitives variables are removed from `wrangler.toml`
- These sensitives variables are uploaded as secrets
- The generated wrangler.toml now contain information of the current environment only

### CODE REVIEW

**Comment by taronaleksanian** [comment]
@Tadjaur that works, only can we run preview workflow only on push / merge on development branch instead of this branch. also we can define change paths for preview workflow too.

**Comment by Tadjaur** [comment]
> @Tadjaur that works, only can we run preview workflow only on push / merge on development branch instead of this branch. also we can define change paths for preview workflow too.

@taronaleksanian what is the name of the development branch? is it `dev` ?

File: .github/workflows/backend-preview.yml [hunk 1]
```diff
@@ -1,91 +1,116 @@
...
+on:
+  workflow_dispatch:
+  push:
+    branches:
+      - andrew_testing
+    paths:
+      - '.github/workflows/backend-preview.yml'
+      - 'packages/validations/**'
+      - 'packages/shared-types/**'
+      - 'server/**'
```
[the remark this answers is not in the document]
Comment (taronaleksanian) [reply .github/workflows/backend-preview.yml:12]: yeah now they're uncommented, thanks

**Review by andrew-bierman — APPROVED** [review]
Nice job :)

**Comment by mikib0** [comment]
> All sensitives variables are removed from `wrangler.toml`
Let's put sensitive variables in `.dev.vars` for local development . Also, we should update readme (?), wrangler.toml.example and add a new `.dev.vars.example`. I can push these updates and it'll then be perfect to go I guess.
Thoughts?

**Comment by mikib0** [comment]
FLet me not block this, I think it's good go

**Comment by andrew-bierman** [comment]
> > All sensitives variables are removed from `wrangler.toml`
> 
> Let's put sensitive variables in `.dev.vars` for local development . Also, we should update readme (?), wrangler.toml.example and add a new `.dev.vars.example`. I can push these updates and it'll then be perfect to go I guess. Thoughts?

I agree we should use dev.vars file for local dev, and then we can keep wrangler toml in git which will make repo easier to spin up.

Merging this in, but agree with the feedback and we should add that

**Comment by mikib0** [comment]
Great 👍. Just realised that I mistakenly closed this btw😅.

**Pull request merged**